aspnetcore/fundamentals/servers/httpsys.md

```diff
@@ -1,179 +1,168 @@
 
-HTTP.sys uses the [HTTP Server API UrlPrefix string formats](https://msdn.microsoft.com/library/windows/desktop/aa364698.aspx).
+   * **.NET Core** &ndash; If the app requires .NET Core, obtain and run the .NET Core installer from [.NET Downloads](https://www.microsoft.com/net/download/windows).
+   * **ASP.NET 4.x** &ndash; If the app requires ASP.NET 4.x, see [.NET Framework: Installation guide](/dotnet/framework/install/) to find installation instructions. Install the required ASP.NET 4.x framework. The installer for the latest ASP.NET 4.x framework can be found at [.NET Downloads](https://www.microsoft.com/net/download/windows).
 
-> [!NOTE]
-> Make sure that you specify the same prefix strings in `UseUrls` or `UrlPrefixes` that you preregister on the server. 
+1. Configure URLs and ports for the app.
 
-### Don't use IIS
+   By default, ASP.NET Core binds to `http://localhost:5000`. To configure URL prefixes and ports, options include using:
 
-Make sure your application isn't configured to run IIS or IIS Express.
+   * [UseUrls](/dotnet/api/microsoft.aspnetcore.hosting.hostingabstractionswebhostbuilderextensions.useurls)
+   * `urls` command-line argument
+   * `ASPNETCORE_URLS` environment variable
+   * [UrlPrefixes](/dotnet/api/microsoft.aspnetcore.server.httpsys.httpsysoptions.urlprefixes)
 
-In Visual Studio, the default launch profile is for IIS Express. To run the project as a console application, manually change the selected profile, as shown in the following screen shot.
+   The following code example shows how to use [UrlPrefixes](/dotnet/api/microsoft.aspnetcore.server.httpsys.httpsysoptions.urlprefixes):
 
-![Select console app profile](httpsys/_static/vs-choose-profile.png)
+   [!code-csharp[](httpsys/sample/Program.cs?name=snippet1&highlight=11)]
 
-## Preregister URL prefixes and configure SSL
+   An advantage of `UrlPrefixes` is that an error message is generated immediately for improperly formatted prefixes.
 
-Both IIS and HTTP.sys rely on the underlying Http.Sys kernel mode driver to listen for requests and do initial processing. In IIS, the management UI gives you a relatively easy way to configure everything. However, you need to configure Http.Sys yourself. The built-in tool for doing that's *netsh.exe*. 
+   The settings in `UrlPrefixes` override `UseUrls`/`urls`/`ASPNETCORE_URLS` settings. Therefore, an advantage of `UseUrls`,`urls`, and the `ASPNETCORE_URLS` environment variable is that it's easier to switch between Kestrel and HTTP.sys. For more information on `UseUrls`, `urls`, and `ASPNETCORE_URLS`, see [Hosting](xref:fundamentals/hosting).
 
-With *netsh.exe* you can reserve URL prefixes and assign SSL certificates. The tool requires administrative privileges.
+   HTTP.sys uses the [HTTP Server API UrlPrefix string formats](https://msdn.microsoft.com/library/windows/desktop/aa364698.aspx).
```

Comment:
Is there a different link we can use? I'd prefer to avoid MSDN.

Reply:
I looked when I was updating the links. Unfortunately, no. I can't find this content anywhere except on MSDN.